.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,198 @@
+          cache-to: type=local,dest=/tmp/.build-cache,mode=max
+
+  unit-test:
+    name: Run unit tests
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Use cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.build-cache
+          key: ${{ runner.os }}-docker-${{ github.sha }}
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: false
+          tags: ${{ env.TEST_TAG }}:builder
+          file: docker/Dockerfile
+          cache-from: type=local,src=/tmp/.build-cache
+          load: true
+          target: builder
+
+      - name: Run unit tests
+        run: |
+          docker-compose -f docker/unit-tests.test.yml \
+            up --exit-code-from=sut
+        env:
+          IMAGE: ${{ env.TEST_TAG }}:builder
+
+  recipe-test:
+    name: Run recipe tests
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Use cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.build-cache
+          key: ${{ runner.os }}-docker-${{ github.sha }}
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: false
+          tags: ${{ env.TEST_TAG }}
+          file: docker/Dockerfile
+          cache-from: type=local,src=/tmp/.build-cache
+          load: true
+
+      - name: Run test recipes on host
+        run: |
+          docker-compose -f docker/recipes.test.yml \
+            up --exit-code-from=sut
+        env:
+          IMAGE: ${{ env.TEST_TAG }}
+
+  uml-test:
+    name: Run UML tests
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Use cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.build-cache
+          key: ${{ runner.os }}-docker-${{ github.sha }}
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: false
+          tags: ${{ env.TEST_TAG }}
+          file: docker/Dockerfile
+          cache-from: type=local,src=/tmp/.build-cache
+          load: true
+
+      - name: Run test recipes using UML backend
+        run: |
+          docker-compose -f docker/recipes-test-uml.yml \
+            up --exit-code-from=sut
+        env:
+          IMAGE: ${{ env.TEST_TAG }}
+
+  publish:
+    name: Publish
+    needs:
+      - unit-test
+      - recipe-test
+      - uml-test
+    if: github.event_name != 'pull_request'
```

Comment:
what's the event which gets triggered when a pull request is merged ooi ?

Reply:
It's "push" in that case.

Reply:
thinking should we not just push a container in all cases?